pkg/build/build_definition_json.go

```diff
@@ -0,0 +1,15 @@
+/*
+  Copyright (c) 2018, Sylabs, Inc. All rights reserved.
+
+  This software is licensed under a 3-clause BSD license.  Please
+  consult LICENSE file distributed with the sources of this project regarding
+  your rights to use or distribute this software.
+*/
+package build
+
+// build_definition_json.go:
+// Provide transformation from JSON -> BuildDefinition
+
+func BuildDefinitionFromJSON(j []byte) BuildDefinition {
+	return BuildDefinition{}
+}
```

Comment:
I'm not totally sure this file is necessary. The structure itself can describe how to encode/decode the JSON, and then we can just use the standard packages?

Reply:
yeah, the JSON std pkg is great for building a JSON from a data structure.

Reply:
I agree that we use the standard JSON package, and we'll definitely tag the struct itself, but I figured just for now we separate it into its own file. Like, the functions that do the (un)marshalling can go in this file. Just so we separate implementation of the `BuildDefinition` data structure from the parsing. Especially since `build_definition_deffile.go` is really big, I don't want to clutter everything into `build_definition.go`

I'm open to changing if the prevailing opinion is different